src/helper.rs

```diff
@@ -50,44 +68,57 @@ mod helper {
     }
 }
 
-#[cfg(feature="winit")]
+#[cfg(feature = "winit")]
 mod helper {
-    use std::sync::Arc;
     use super::*;
+    use std::sync::Arc;
     use winit::event::{Event, WindowEvent};
     use winit::event_loop::{ControlFlow, EventLoop};
     use winit::window::Window;
 
     pub use winit;
 
-    pub fn game_loop<G, U, R, H, T>(event_loop: EventLoop<T>, window: Arc<Window>, game: G, updates_per_second: u32, max_frame_time: f64, mut update: U, mut render: R, mut handler: H) -> !
-        where G: 'static,
-              U: FnMut(&mut GameLoop<G, Time, Arc<Window>>) + 'static,
-              R: FnMut(&mut GameLoop<G, Time, Arc<Window>>) + 'static,
-              H: FnMut(&mut GameLoop<G, Time, Arc<Window>>, &Event<'_, T>) + 'static,
-              T: 'static,
+    pub fn game_loop<G, U, R, H, T>(
+        event_loop: EventLoop<T>,
+        window: Arc<Window>,
+        game: G,
+        updates_per_second: u32,
+        max_frame_time: f64,
+        mut update: U,
+        mut render: R,
+        mut handler: H,
+    ) -> Result<(), impl std::error::Error>
```

Comment:
This should be the concrete `EventLoopError` from `winit` instead of an existential type.

Reply:
Thanks for the review!